tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/api/validation/RightToWorkValidatorTest.java

```diff
@@ -195,21 +207,121 @@ void shouldNotReturnErrorWhenNotVisaIssuedAndVisaValidTo() {
     // Given.
     RightToWorkDTO dto = new RightToWorkDTO();
     dto.setVisaValidTo(LocalDate.now());
+    PersonDTO personDTO = new PersonDTO();
+    personDTO.setRightToWork(dto);
+
+    // When.
+    List<FieldError> fieldErrors = validator.validateForBulk(dto, dto.getId());
+
+    // Then
+    assertThat("should not return errors", fieldErrors.size(), is(0));
+  }
+
+  @Test
+  void shouldNotReturnErrorWhenVisaValidToIsAfterDbVisaIssuedDate() {
+    // Given.
+    RightToWork dbdto = new RightToWork();
+    dbdto.setVisaIssued(LocalDate.now());
+
+    RightToWorkDTO dto = new RightToWorkDTO();
+    dto.setVisaValidTo(LocalDate.now().plusDays(10));
+
+    PersonDTO personDTO = new PersonDTO();
+    personDTO.setRightToWork(dto);
+    personDTO.setId(1L);
+
+    Person person = new Person();
+    person.setId(1L);
+    person.setRightToWork(dbdto);
+    // When.
+
+    when(personRepository.findPersonById(1L)).thenReturn(java.util.Optional.of(person));
+
+    List<FieldError> fieldErrors = validator.validateForBulk(dto, dto.getId());
+
+    // Then
+    assertThat("should not return errors", fieldErrors.size(), is(0));
+  }
+
+  @Test
+  void shouldNotReturnErrorWhenVisaIssuedDateToIsBeforeDbVisaValidToDate() {
+    // Given.
+    RightToWork dbdto = new RightToWork();
+    dbdto.setVisaValidTo(LocalDate.now().plusDays(10));
+
+    RightToWorkDTO dto = new RightToWorkDTO();
+    dto.setVisaIssued(LocalDate.now());
 
+    PersonDTO personDTO = new PersonDTO();
+    personDTO.setRightToWork(dto);
+    personDTO.setId(1L);
+
+    Person person = new Person();
+    person.setId(1L);
+    person.setRightToWork(dbdto);
     // When.
-    List<FieldError> fieldErrors = validator.validateForBulk(dto);
+
+    when(personRepository.findPersonById(1L)).thenReturn(java.util.Optional.of(person));
```

Comment:
The `Optional` should be imported `import java.util.Optional;` and then delete ~java.util.~.